sycl-fusion/passes/cleanup/Cleanup.cpp

```diff
@@ -74,7 +72,7 @@ static Function *createMaskedFunction(const BitVector &Mask, Function *F) {
     // Copy metadata.
     SmallVector<std::pair<unsigned, MDNode *>> MDs;
     F->getAllMetadata(MDs);
-    for (auto MD : MDs) {
+    for (auto &MD : MDs) {
```

Comment:
Please note that I haven't thoroughly reviewed the rest of `for (auto ...)` loops (because the patch is quite huge), so there might be similar excessive copies somewhere else